src/dbmanager.cpp

```diff
@@ -277,20 +278,22 @@ bool DBManager::updateNote(NoteData* note)
     int id = note->id();
     qint64 epochTimeDateModified = note->lastModificationdateTime().toMSecsSinceEpoch();
     QString content = note->content()
-            .replace("'","''")
+            .replace(QLatin1String("'"), QLatin1String("''"))
             .replace(QChar('\x0'), emptyStr);
     QString fullTitle = note->fullTitle()
-            .replace("'","''")
+            .replace(QLatin1String("'"), QLatin1String("''"))
```

Comment:
Oh btw, I think this replacement should be removed now. `bindValue` should already do any needed quoting, so now this will be duplicating single quotes.

Reply:
Is there anything said about this in documentation ?

Reply:
Indeed. That's what bindValue is doing so no point in keep these. It does lead to duplication of single quotes every time the note is saved.

Reply:
> Is there anything said about this in documentation ?

I couldn't immediate find it in Qt's documentation, but bound values are a common feature in SQL databases so Qt is probably just forwarding these functions to the underlying SQL API. It could even avoid the need for escaping entirely since the values are no longer part of the query. So it's good for security as well as performance.

Reply:
From the `QSqlQuery` docs:

> QSqlQuery supports prepared query execution and the binding of parameter values to placeholders. Some databases don't support these features, so for those, Qt emulates the required functionality. For example, the Oracle and ODBC drivers have proper prepared query support, and Qt makes use of it; but for databases that don't have this support, Qt implements the feature itself, e.g. by replacing placeholders with actual values when a query is executed

And SQLite does support bindValues
https://sqlite.org/c3ref/bind_blob.html